packages/framework/src/evaluate/binding.ts

```diff
@@ -105,7 +106,7 @@ export const createBindingAtom = <T = unknown>(
           : undefined,
         formatter: DateFormatter(),
       },
-      context,
+      context: { ...context, device: get(deviceAtom) },
```

Comment:
Can we only include this if device is part of the expression to avoid creating extra dependencies? It should come before context also